Microsoft.Identity.Web/Resource/ScopesRequiredByWebAPIExtension.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Text;
+
+namespace Microsoft.Identity.Web.Resource
+{
+    public static class ScopesRequiredByWebAPIExtension
+    {
+        /// <summary>
+        /// When applied to a Controller, verifies that the user authenticated in the Web API has any of the
+        /// accepted scopes
+        /// </summary>
+        /// <param name="acceptedScopes">Scopes accepted by this API</param>
+        public static void VerifyUserHasAnyAcceptedScope(this HttpContext context, IEnumerable<string> acceptedScopes)
```

Comment:
A good time to start writing unit tests? :D

Reply:
Thanks @bgavrilMS .
Let's dicuss tomorrow for the UT.